search/search_engine_pylib/api.cpp

```diff
@@ -0,0 +1,254 @@
+// features_collector_tool and search_quality_tool. Need to replace
+// both tools by python scripts that use this library.
+
+void DidDownload(storage::TCountryId const & /* countryId */,
+                 shared_ptr<platform::LocalCountryFile> const & /* localFile */)
+{
+}
+
+bool WillDelete(storage::TCountryId const & /* countryId */,
+                shared_ptr<platform::LocalCountryFile> const & /* localFile */)
+{
+  return false;
+}
+
+unique_ptr<storage::Storage> g_storage;
+
+void Init(string const & resource_path, string const & mwm_path)
+{
+  auto & platform = GetPlatform();
+
+  string countriesFile = COUNTRIES_FILE;
+
+  if (!resource_path.empty())
+  {
+    platform.SetResourceDir(resource_path);
+    countriesFile = my::JoinFoldersToPath(resource_path, COUNTRIES_FILE);
+  }
+
+  if (!mwm_path.empty())
+    platform.SetWritableDirForTests(mwm_path);
+
+  classificator::Load();
+
+  g_storage = make_unique<storage::Storage>(countriesFile, mwm_path);
+  g_storage->Init(&DidDownload, &WillDelete);
+}
+
+struct LatLon
+{
+  LatLon() = default;
+  LatLon(double lat, double lon) : m_lat(lat), m_lon(lon) {}
+
+  string ToString() const
+  {
+    ostringstream os;
+    os << "lat:" << m_lat << ", lon:" << m_lon;
+    return os.str();
+  }
+
+  double m_lat = 0.0;
+  double m_lon = 0.0;
+};
+
+struct Mercator
+{
+  Mercator() = default;
+  Mercator(double x, double y) : m_x(x), m_y(y) {}
+  Mercator(m2::PointD const & m): m_x(m.x), m_y(m.y) {}
+
+  string ToString() const
+  {
+    ostringstream os;
+    os << "x: " << m_x << ", y: " << m_y;
+    return os.str();
+  }
+
+  double m_x = 0.0;
+  double m_y = 0.0;
+};
+
+struct Viewport
+{
+  Viewport() = default;
+  Viewport(Mercator const & min, Mercator const & max) : m_min(min), m_max(max) {}
+
+  string ToString() const
+  {
+    ostringstream os;
+    os << "[" << m_min.ToString() << ", " << m_max.ToString() << "]";
+    return os.str();
+  }
+
+  Mercator m_min;
+  Mercator m_max;
+};
+
+struct Params
+{
+  string ToString() const
+  {
+    ostringstream os;
+    os << m_query << ", " << m_locale << ", " << m_latLon.ToString() << ", "
+       << m_viewport.ToString();
+    return os.str();
+  }
+
+  string m_query;
+  string m_locale;
+  LatLon m_latLon;
+  Viewport m_viewport;
+};
+
+struct Result
+{
+  Result() = default;
+
+  Result(search::Result const & r)
+  {
+    m_name = r.GetString();
+    m_address = r.GetAddress();
+    m_hasCenter = r.HasPoint();
+    if (m_hasCenter)
+      m_center = r.GetFeatureCenter();
+  }
+
+  string ToString() const
+  {
+    ostringstream os;
+    os << m_name << " [ " << m_address;
+    if (m_hasCenter)
+      os << ", " << m_center.ToString();
+    os << " ]";
+    return os.str();
+  }
+
+  string m_name;
+  string m_address;
+  bool m_hasCenter = false;
+  Mercator m_center;
+};
+
+struct SearchEngineProxy
+{
+  SearchEngineProxy()
+  {
+    CHECK(g_storage.get() != nullptr, ("init() is not called."));
```

Comment:
This message is displayed only when SearchEngine is created on the python side. Therefore, it would be more meaningful to write python-side function name, not C++-side.